ribbon-core/src/main/java/com/netflix/client/config/CommonClientConfigKey.java

```diff
@@ -248,20 +244,20 @@ public Class type() {
     
     private final String configKey;
     private final Class<T> type;
-    private T defaultValue;
+    private final T defaultValue;
 
-    @SuppressWarnings("unchecked")
     protected CommonClientConfigKey(String configKey) {
         this(configKey, null);
     }
 
+    @SuppressWarnings("unchecked")
     protected CommonClientConfigKey(String configKey, T defaultValue) {
         this.configKey = configKey;
         Type superclass = getClass().getGenericSuperclass();
-        checkArgument(superclass instanceof ParameterizedType,
-                "%s isn't parameterized", superclass);
-        Type runtimeType = ((ParameterizedType) superclass).getActualTypeArguments()[0];
-        type = (Class<T>) TypeToken.of(runtimeType).getRawType();
+        if (!(superclass instanceof ParameterizedType)) {
+                throw new IllegalArgumentException(superclass + " isn't parameterized");
+        }
+        this.type = (Class<T>) ((ParameterizedType) superclass).getActualTypeArguments()[0];
```

Comment:
This is probably one of the only changes that I am not completely confident in, looks like the tests were happy with this change but not sure of other usages not covered in test cases